### PR TITLE
Run all tests on new LoanFactory, remove LoanFactory3.sol

### PR DESCRIPTION
**Pull request by vanruch**
I skipped some TrueFi tests because it now becomes impossible to create loans on it - it lacks proFormaLiquidRatio function

### CODE REVIEW

**Comment by vanruch** [comment]
This breaks one more integration test

File: test/truefi2/lines-of-credit/TimeAveragedBaseRateOracle.test.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ describe('TimeAveragedBaseRateOracle', () => {
     it('returns false if cooldown is on', async () => {
       await timeBaseRateOracle.update()
       expect(await timeBaseRateOracle.isOffCooldown()).to.be.false
-      await timeTravel(provider, COOLDOWN_TIME - 1)
+      await timeTravel(provider, COOLDOWN_TIME - 10)
```
Comment (MiksuJak) [inline test/truefi2/lines-of-credit/TimeAveragedBaseRateOracle.test.ts:151]: 👍

**Review by MiksuJak — APPROVED** [review]

File: test/truefi2/LoanFactory2.test.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import { expect, use } from 'chai'
```
Comment (yuchenlintt) [inline test/truefi2/LoanFactory2.test.ts:1]: Confirming I diffed against LoanFactory3.test.ts, with no changes apart from expected naming.

File: test/utils/setupTruefi2.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ import {
 import { Wallet } from 'ethers'
```
Comment (yuchenlintt) [inline test/utils/setupTruefi2.ts:20]: Confirming I diffed against setupTruefi2WithLoanFactory3.ts.

Mostly the same apart from naming and expected changes.

File: test/utils/connectMockContract.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
+// Connect to mock contract without deploying it
+// We should add this to Waffle and remove from here
```
Comment (yuchenlintt) [inline test/utils/connectMockContract.ts:2]: +1

This is pretty awesome! Scary to review as part of a large PR though. Nothing jumps out to me on a cursory read-through, but we should take a closer look post-merge to understand what's going on here.

**Review by yuchenlintt — APPROVED** [review]
LGTM. I diffed all LoanFactory3-related files against their LoanFactory2 counterparts and confirmed they're mostly identical except name changes.

I have some concerns about introducing `connectMockContract` as part of this PR, but since this is a large expected change, we should fast-track the merge and review that file post-merge. In the future let's extract this kind of code into a separate PR.

File: test/utils/connectMockContract.ts (hunk 4 above)
Comment (vanruch) [reply test/utils/connectMockContract.ts:2]: It's mostly copied code from Waffle

Comment (vanruch) [reply test/utils/connectMockContract.ts:2]: The only thing changed is removing of mock contract deployment

Comment (yuchenlintt) [reply test/utils/connectMockContract.ts:2]: Thanks for the background! I guess this is also a test util, so the scope of possible damage isn't as great as modifying Solidity code. LGTM.

**Pull request merged**